torch/csrc/distributed/c10d/init.cpp

```diff
@@ -225,6 +225,15 @@ PyObject* c10d_init(PyObject* _unused, PyObject* noargs) {
           R"(
 Returns:
     A list of ``torch.Tensor``. Each tensor in the list corresponds to a parameter.
+)")
+      .def(
+          "is_the_last_bucket_to_allreduce",
+          &::c10d::GradBucket::isTheLastBucketToAllreduce,
+          py::call_guard<py::gil_scoped_release>(),
+          R"(
+Returns:
+    Whether this bucket is the last bucket to allreduce in an iteration.
+    This also means that this bucket corresponds to the first few layers in the forward pass.
```

Comment:
nit: This may not necessarily be true if reducer rebuilds the bucket in true gradient ready order

Reply:
True gradient ready order is only possible when `tensor_indices` is provided in `compute_bucket_assignment_by_size`. This arg seems to be always empty in the codebase.

Reply:
I think it is provided in `rebuild_buckets` as `rebuilt_param_indices_`, which is how the buckets are rebuilt in true gradient ready order (cc @yanlizhao). By default, when it is not provided then we assume gradient ready order is reverse of parameters.